### PR TITLE
[Feat]: Implemented Watch History

### PR DESCRIPTION
**Pull request by parthpandyappp**
- User can see the watch history in tabular format.
- User can `removeAll` watch history in one go.
- User can `remove` specific video from watch history.

### CODE REVIEW

**Review by ravikumar0403 — APPROVED** [review]
code looks clean
file structure is good
design looks consistent

File: src/contexts/GenericDataProvider.js [hunk 1]
```diff
@@ -34,9 +34,26 @@ function GenericDataProvider({ children }) {
...
+            const response = await axios({
+                method: "GET",
+                headers: {
+                    authorization: encodedToken, // passing token as an authorization header
+                },
+                url: "/api/user/history",
+            });
+            dispatch({ type: "SET_HISTORY_DATA", payload: response.data.history });
+        } catch (error) {
+            console.log(error);
```
Comment (vaibhav18matere) [inline src/contexts/GenericDataProvider.js:49]: remove console.log

File: src/contexts/GenericDataProvider.js [hunk 2]
```diff
@@ -34,9 +34,26 @@ function GenericDataProvider({ children }) {
         }
     };
 
+    const getHistoryData = async () => {
+        try {
+            const encodedToken = localStorage.getItem("token");
+            const response = await axios({
+                method: "GET",
+                headers: {
+                    authorization: encodedToken, // passing token as an authorization header
```
Comment (vaibhav18matere) [inline src/contexts/GenericDataProvider.js:43]: remove commented lines if not needed for further use

File: src/helper-functions/watchhistory-helpers.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            method: "DELETE",
+            headers: {
+                authorization: encodedToken,
+            },
+            url: `/api/user/history/all`,
+        });
+
+        boolFunc();
+    } catch (error) {
+        console.log(error);
```
Comment (vaibhav18matere) [inline src/helper-functions/watchhistory-helpers.js:57]: remove this or show error on UI

**Review by vaibhav18matere — APPROVED** [review]
looks good; just check small improvements given below

File: src/contexts/GenericDataProvider.js (hunk 2 above)
Comment (parthpandyappp) [reply src/contexts/GenericDataProvider.js:43]: Comments are for watchers who want to understand the code.

File: src/helper-functions/watchhistory-helpers.js (hunk 3 above)
Comment (parthpandyappp) [reply src/helper-functions/watchhistory-helpers.js:57]: Noted :100:

**Comment by parthpandyappp** [comment]
Thanks @vaibhav18matere for the PR review :+1:

**Pull request merged**